prov/mlx/src/mlx_av.c

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright (c) 2016 Intel Corporation. All rights reserved.
+ *
+ * This software is available to you under a choice of one of two
+ * licenses.  You may choose to be licensed under the terms of the GNU
+ * General Public License (GPL) Version 2, available from the file
+ * COPYING in the main directory of this source tree, or the
+ * OpenFabrics.org BSD license below:
+ *
+ *     Redistribution and use in source and binary forms, with or
+ *     without modification, are permitted provided that the following
+ *     conditions are met:
+ *
+ *      - Redistributions of source code must retain the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer.
+ *
+ *      - Redistributions in binary form must reproduce the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer in the documentation and/or other materials
+ *        provided with the distribution.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include "mlx.h"
+
+static int mlx_av_write_event(
+				struct mlx_av *av, uint64_t data,
+				int err, void *context)
+{
+	struct fi_eq_err_entry entry;
+	size_t size;
+	uint64_t flags;
+
+	entry.fid = &(av->av.fid);
+	entry.context = context;
+	entry.data = data;
+
+	if (err) {
+		entry.err = err;
+		size = sizeof(struct fi_eq_err_entry);
+		flags = UTIL_FLAG_ERROR;
+	} else {
+		size = sizeof(struct fi_eq_entry);
+		flags = 0;
+	}
+
+	fi_eq_write(
+		&(av->eq->eq_fid), FI_AV_COMPLETE,
+		&entry, size, flags);
+	return FI_SUCCESS;
+}
+
+static int mlx_av_remove(
+			struct fid_av *fi_av, fi_addr_t *fi_addr, size_t count,
+			uint64_t flags)
+{
+	struct mlx_av *av;
+	int i;
+
+	av = container_of(fi_av, struct mlx_av, av);
+	if ((av->async) && (!av->eq))
+	{
+		return -FI_ENOEQ;
+	}
+
+	for (i = 0; i < count; ++i)
+	{
+		ucp_ep_destroy((ucp_ep_h)(fi_addr[i]));
+	}
```

Comment:
nit: here and in other places, please follow the Linux kernel coding conventions.  I.e. put the open brace at the end of the if/for/while lines.